### ISSUE TITLE
Unit test open flow controller syncFlows()

### ISSUE THREAD

**Issue by andrewsykim** [k-vswitch/k-vswitch#7]
Some unit tests for syncFlows() would be nice. One approach is to create a kubernetes fake client, generate the resources we are watching for (vswitchconfig, pods and network policies) and validate that the flows buffer that would be passed into `ovs-ofctl replace-flows`. It doesn't actually validate that the flows are configured properly but I think it's still a good improvement from what we have now.

**Comment by mdlayher** [comment]
Yep. This is the same approach I took with go-openvswitch. You could potentially do more interesting things with mininet and the like, but IMO this is a good idea for unit tests.

**Comment by zanetworker** [comment]
For a real e2e test, I would even consider using something like **Kind** with a modified docker image (OVS installed), deploy k-vswitch, and run the test cases. As @mdlayher said, mininet could be used, it might get you closer to the real thing but not the real thing (i.e., not kubernetes).

**Comment by andrewsykim** [comment]
Agreed, we should have a separate issue for e2e
